### PR TITLE
Log view and authentication

### PR DESCRIPTION
**Pull request by fabianfrz**
allow authentication, user management, pass through log to frontend (authenticated only)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,6 +89,9 @@ set(THREADS_PREFER_PTHREAD_FLAG TRUE)
 find_package(Threads REQUIRED)
 
 # fetch libwebsockets and add library
+# uncomment to disable TLS
+set(LWS_WITH_SSL OFF)
```
Comment (fabianfrz) [inline CMakeLists.txt:93]: need to comment that out again

File: util/WebSocketServer.cpp [hunk 2]
```diff
@@ -115,6 +161,7 @@ WebSocketServer::WebSocketServer( int port, const string certPath, const string&
         info.ssl_cert_filepath        = NULL;
         info.ssl_private_key_filepath = NULL;
     }
+#endif
```
Comment (fabianfrz) [inline util/WebSocketServer.cpp:164]: remove that again

File: util/WebSocketServer.cpp [hunk 3]
```diff
@@ -172,24 +220,24 @@ void WebSocketServer::onErrorWrapper( int socketID, const string& message )
...
     // Push this onto the buffer. It will be written out when the socket is writable.
-    this->connections[socketID]->buffer.push_back( data.c_str() );
+    this->connections[socketID]->push_to_buffer( data );
 }
 
 void WebSocketServer::broadcast(const string& data )
 {
     for(auto & connection : this->connections) {
-        this->send(connection.first, data);
+        //this->send(connection.first, data);
```
Comment (fabianfrz) [inline util/WebSocketServer.cpp:229]: remove the comment sign again - forgot to remove it again

**Comment by isemann** [comment]
Great job, thanks.

**Pull request merged**